internal/xds/xdsclient/xdsresource/unmarshal_eds.go

```diff
@@ -204,3 +206,40 @@ func parseEDSRespProto(m *v3endpointpb.ClusterLoadAssignment) (EndpointsUpdate,
 	}
 	return ret, nil
 }
+
+func validateAndConstructMetadata(metadataProto *v3corepb.Metadata) (Metadata, error) {
+	metadata := make(map[string]MetadataValue)
+	if metadataProto == nil {
+		return Metadata{Metadata: metadata}, nil
+	}
+	// First go through TypedFilterMetadata.
+	for key, anyProto := range metadataProto.GetTypedFilterMetadata() {
+		converter := ConverterForType(anyProto.GetTypeUrl())
+		// Ignore types we don't have a converter for.
+		if converter == nil {
+			continue
+		}
+
+		val, err := converter.Convert(anyProto.GetValue())
+		if err != nil {
+			// If the converter fails, nack the whole resource.
+			return Metadata{}, fmt.Errorf("metadata parser for key %q and type %q failed: %v", key, anyProto.GetTypeUrl(), err)
+		}
+		metadata[key] = val
+	}
+
+	// Process FilterMetadata for any keys not already handled.
+	for key, structProto := range metadataProto.GetFilterMetadata() {
+		_, exists := metadata[key]
+		// Skip keys already added from TyperFilterMetadata.
+		if exists {
+			continue
+		}
```

Comment:
The assignment and the conditional can be folded into a single line.

```golang
		// Skip keys already added from TyperFilterMetadata.
		if _, exists := metadata[key]; exists {
			continue
		}
```

Or you could even do:
```golang
		// Skip keys already added from TyperFilterMetadata.
		if metadata[key] != nil {
			continue
		}
```